src/core/graph.go

```diff
@@ -203,19 +200,15 @@ func (graph *BuildGraph) SubrepoOrDie(name string) *Subrepo {
 
 // AllTargets returns a consistently ordered slice of all the targets in the graph.
 func (graph *BuildGraph) AllTargets() BuildTargets {
-	targets := BuildTargets{}
-	graph.targets.ForEach(func(k, v interface{}) bool {
-		targets = append(targets, v.(*BuildTarget))
-		return true
-	})
+	targets := graph.targets.Values()
 	sort.Sort(targets)
 	return targets
 }
 
 // PackageMap returns a copy of the graph's internal map of name to package.
 func (graph *BuildGraph) PackageMap() map[string]*Package {
 	packages := map[string]*Package{}
-	graph.packages.ForEach(func(k, v interface{}) bool {
+	graph.packages.ForEachLocked(func(k, v interface{}) bool {
```

Comment:
Why did this have to change?

Reply:
It didn't have to, I changed it with the other one (then later changed that too).
I think I will put it back for now. Looks like it might be called concurrently (although I'm a bit nervous about the way this gets called now I look at it...).